src/GafferScene/RendererAlgo.cpp

```diff
@@ -1109,9 +1109,27 @@ void applyCameraGlobals( IECoreScene::Camera *camera, const IECore::CompoundObje
 	}
 
 	const BoolData *depthOfFieldData = globals->member<BoolData>( "option:render:depthOfField" );
-	bool depthOfField = depthOfFieldData && depthOfFieldData->readable();
+	/*if( !camera->hasDepthOfField() && depthOfFieldData )
+	{
+		camera->setDepthOfField( depthOfFieldData->readable() );
+	}*/
+	// \todo - switch to the form above once we have officially added the depthOfField parameter to Cortex.
+	// The plan then would be that the renderer backends should respect camera->getDepthOfField.
+	// For the moment we bake into fStop instead
+	bool depthOfField = false;	
+	if( depthOfFieldData )
+	{
+		// First set from render globals
+		depthOfField = depthOfFieldData->readable();
+	}
+	if( camera->parameters().find( "depthOfField" ) != camera->parameters().end() )
```

Comment:
We should be able to combine the two map lookups and avoid a potential null dereference as follows :

```
if( const BoolData *d = camera->parametersData()->member<BoolData>( "depthOfField" ) )
{
    depthOfField = d->readable();
}
```